### PR TITLE
feat(core/list/space-list/popover): theming changes

### PR DESCRIPTION
**Pull request by torgeadelin**
## Description

The scope of the changes in this pull request are to add css theming variables to the input core component.
[SPARK-230827](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-230827)

## Motivation and Context

The changes in this PR are to meet presented requirements for the new theming.

## Screenshots:
Live user testing was done to confirm that the changes properly impacted the components without impacting existing use cases.

### Popover
Before:
image

After:
image

### Space List
Before:
image

After:
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Other... Please describe:

- `mixin scrollbar()` has been changed slightly (in addition to the theming changes).
The first variable has been changed to just `border-color` instead of providing the whole border css value (eg 4px solid ...). All instances of `@include` were having the exact same value.
- for the `Popover` component, I replaced the border trick using `box-shadow` with proper `border`. The reason for this is because if you have a `List` inside the popover, the hover of the list-items would cover the border. Using `border` adds the border outside (outset instead of inset) of the container, fixing this issue.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by torgeadelin** [comment]
Thanks @futurethang, I'll ask somebody from my team to review it.

@InteractiveTimmy @robstax could you have a look at this when you get the chance?

File: core/scss/tools/mixins/scrollbar.scss [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+/// @param {Box-Shadow} $box-shadow - Scrollbar Box-Shadow
+/// @param {Color} $border-color-css-var - CSS variable color for border color
+/// @param {Color} $background-css-var - CSS variable color for background
+/// @param {Color} $hover-background-css-var - CSS variable for Scrollbar's Hover Background
+/// @param {Color} $hover-background - Scrollbar's Hover Background
+
 
 @mixin scrollbar(
-  $border,
+  $border-color,
```
Comment (InteractiveTimmy) [inline core/scss/tools/mixins/scrollbar.scss:17]: Good refactor with `$border` to `$border-color`

File: core/scss/tools/mixins/scrollbar.scss [hunk 2]
```diff
@@ -26,16 +35,20 @@
 
   &::-webkit-scrollbar-thumb {
     background: $foreground;
-    border: $border;
+    background: var($foreground-css-var);
```
Comment (InteractiveTimmy) [inline core/scss/tools/mixins/scrollbar.scss:38]: This needs a default, right?

```scss
background: var($foreground-css-var, $foreground);
```

**Review by InteractiveTimmy — CHANGES_REQUESTED** [review]
Small change request, If all tests pass within reason, I'll approve after changes are made.

File: core/scss/tools/mixins/scrollbar.scss (hunk 2 above)
Comment (torgeadelin) [reply core/scss/tools/mixins/scrollbar.scss]: good catch!

**Review by InteractiveTimmy — DISMISSED** [review]
LGTM

**Review by yjlo — APPROVED** [review]
LGTM

**Pull request merged**